madeofhuman/maintenance-react
# #161062099 login user on signup and refactor forms to fix stale data display

#### What does this PR do?
- logs user in on successful signup
- refactors update component to fix the issue of stale data being displayed on the update form
- refactors error messages to be human-readable and return appropriate messages

#### Description of Task to be completed?
- log user in on successful signup
- refactor update component to fix the issue of stale data being displayed on the update form
- refactor error messages to be human-readable and return appropriate messages

#### How should this be manually tested?
- visit https://maintenance-react-staging.herokuapp.com and click the signup button to create an account. You should be logged in on successful account creation
- visit https://maintenance-react-staging.herokuapp.com and click the button to update a request with a status of 'in-review'. The form should display the correct request

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#161062099](https://www.pivotaltracker.com/story/show/161062099)
[#161081477](https://www.pivotaltracker.com/story/show/161081477)
[#161041551](https://www.pivotaltracker.com/story/show/161041551)

#### Screenshots
N/A

#### Questions:
N/A

[Delivers #161062099]
[Delivers #161081477]
[Delivers #161041551]
